plugins/statistics.koplugin/main.lua

```diff
@@ -1,32 +1,41 @@
-local Widget = require("ui/widget/widget")
-local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local DocSettings = require("docsettings")
+local Event = require("ui/event")
+local InfoMessage = require("ui/widget/infomessage")
 local KeyValuePage = require("ui/widget/keyvaluepage")
-local UIManager = require("ui/uimanager")
-local Screen = require("device").screen
+local MultiInputDialog = require("ui/widget/multiinputdialog")
+local ReaderProgress = require("readerprogress")
+local ReadHistory = require("readhistory")
+local SQ3 = require("lua-ljsqlite3/init")
 local TimeVal = require("ui/timeval")
-local DataStorage = require("datastorage")
+local UIManager = require("ui/uimanager")
+local Widget = require("ui/widget/widget")
 local lfs = require("libs/libkoreader-lfs")
-local DEBUG = require("dbg")
-local T = require("ffi/util").template
-local joinPath = require("ffi/util").joinPath
-local _ = require("gettext")
+local logger = require("logger")
 local util = require("util")
-local tableutil = require("tableutil")
-local ReadHistory = require("readhistory")
-local DocSettings = require("docsettings")
-local ReaderProgress = require("readerprogress")
-local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local _ = require("gettext")
+local joinPath = require("ffi/util").joinPath
+local Screen = require("device").screen
+local T = require("ffi/util").template
 
--- a copy of page_max_read_sec
-local page_max_time
+local statistics_dir = DataStorage:getDataDir() .. "/statistics/"
+local db_location = DataStorage:getSettingsDir() .. "/book.sqlite3"
 
 local ReaderStatistics = Widget:extend{
-    last_time = nil,
-    page_min_read_sec = 5,
-    page_max_read_sec = 90,
-    current_period = 0,
-    pages_current_period = 0,
+    page_min_read_sec = 1,
+    page_max_read_sec = 180,
```

Comment:
Better, but if that was the default it's still too short because it caused bug reports about stats not working. The default should then be something more like two or three minutes. Even if you did put down your reader for a minute and "forgot", is it really so bad if the device counts an extra minute occasionally?

I can totally see where you're coming from with 15 minutes when you're writing down your thoughts on what you're reading, looking things up in reference works, double-checking calculations, etc., but that wouldn't do as a default. ;-)